src/jit/target.h

```diff
@@ -2009,32 +2013,27 @@ inline bool         isValidIntArgReg(regNumber reg)
 
 //-------------------------------------------------------------------------------------------
 // genRegArgNext:
-//     Given a register that is an integer argument register 
-//     returns the next integer argument register 
+//     Given a register that is an integer or floating point argument register 
+//     returns the next argument register 
 //
 regNumber           genRegArgNext(regNumber argReg);
 
-#if !defined(_TARGET_X86_)
-
 //-------------------------------------------------------------------------------------------
 // isValidFloatArgReg:
 //     Returns true if the register is a valid floating-point argument register 
 //
 inline bool         isValidFloatArgReg(regNumber reg)
 {
-    return (reg >= FIRST_FP_ARGREG) && (reg <= LAST_FP_ARGREG);
+    if (reg == REG_NA)
```

Comment:
I'm a little curious why you're optimizing for this case. Is it common enough to merit it? And it will require an extra check any time it's called with an integer register.